File: server/Models/VaultKeep.cs
namespace keepr.Models;

public class VaultKeep : RepoItem<int>
{
  public int KeepId { get; set; }
  public int VaultId { get; set; }
  public string CreatorId { get; set; }
}

public class Kept : Keep
{
  public int VaultKeepId { get; set; }

}

File: client/src/models/VaultKeep.js
import { DbItem } from "./DbItem";
import { Keep } from "./Keep";

export class VaultKeep extends DbItem{
  /**
   * @param {{ keepId: Number; vaultId: Number; creatorId: String; }} data
   */
  constructor(data){
    super(data)
    this.keepId = data.keepId
    this.vaultId = data.vaultId
    this.creatorId = data.creatorId
  }
}

export class Kept extends Keep{
  /**
   * @param {{ vaultKeepId: Number; vaultKeepCreatorId: String; }} data
   */
  constructor(data){
    super(data)
    this.vaultKeepId = data.vaultKeepId
    this.vaultKeepCreatorId = data.vaultKeepCreatorId
  }
}